Editor/AGS.Editor/ImportExport.cs

```diff
@@ -311,6 +311,38 @@ public static CompileMessages ImportOldEditorDatFile(string fileName, Game game,
             return importErrors;
         }
 
+        private static List<Script> AddImportedScriptModule(Encoding defEncoding, byte[] author, byte[] description, byte[] name, string version, byte[] scriptBytes, byte[] headerBytes, int uniqueKey, int encodingCP = 0)
+        {
+            Encoding enc = defEncoding;
+            try
+            {
+                if (encodingCP > 0)
+                    enc = Encoding.GetEncoding(encodingCP);
+            }
+            catch (ArgumentException) { }
+
+            List<Script> scriptsImported = new List<Script>();
+            Script header = new Script(null, enc.GetString(headerBytes), enc.GetString(name),
+                enc.GetString(description), enc.GetString(author), version, uniqueKey, true);
+            Script mainScript = new Script(null, enc.GetString(scriptBytes), enc.GetString(name),
+                enc.GetString(description), enc.GetString(author), version, uniqueKey, false);
+            scriptsImported.Add(header);
+            scriptsImported.Add(mainScript);
+            return scriptsImported;
+        }
+
+        public static List<Script> AddImportedScriptFromAscAsh(Encoding defEncoding, string scriptName, int uniqueKey, byte[] scriptBytes, byte[] headerBytes)
+        {
+            byte[] b_empty = defEncoding.GetBytes(string.Empty);
+
+            byte[] author = b_empty;
+            byte[] description = b_empty;
+            string version = string.Empty;
+            byte[] name = defEncoding.GetBytes(scriptName);
+
+            return AddImportedScriptModule(defEncoding, author, description, name, version, scriptBytes, headerBytes, uniqueKey);

```

Comment:
I know this is a draft, but I'd like to comment on these 2 functions right away.
I think it's better to have the "final" function with convenient argument types, which is "string" in this case, and if input data is in different type (like `byte[]`), then convert that to the final type. In this code this is done "backwards", which makes string converted to byte[] and then byte[] converted to string for no good reason.

Reply:
Another thing that I might propose is to make a struct for holding this module's "meta info", create that struct from input, and pass into the "final" function that imports the module.

Reply:
Did a push that tries to address this comment: https://github.com/adventuregamestudio/ags/compare/d9a0902208c100097560f56739fe723bcef2b63b..4cf2490ae863ccb539d541976460d563ce46a62a

the code became a bit longer but I like to break things in separate ways so I can debug it more easily. I can try to reduce the number of lines if needed.